python/cudf/cudf/tests/test_list.py

```diff
@@ -112,3 +112,55 @@ def test_len(data):
     got = gsr.list.len()
 
     assert_eq(expect, got, check_dtype=False)
+
+
+@pytest.mark.parametrize(
+    ("data", "idx"),
+    [
+        ([[1, 2, 3], [3, 4, 5], [4, 5, 6]], [[0, 1], [2], [1, 2]]),
+        ([[1, 2, 3], [3, 4, 5], [4, 5, 6]], [[1, 2, 0], [1, 0, 2], [0, 1, 2]]),
+        ([[1, 2, 3], []], [[0, 1], []]),
+        ([[1, 2, 3], [None]], [[0, 1], []]),
+        ([[1, None, 3], None], [[0, 1], []]),
+    ],
+)
+def test_take(data, idx):
+    ps = pd.Series(data)
+    gs = cudf.from_pandas(ps)
+
+    pdf = pd.DataFrame({"data": ps, "idx": idx})
+    expected = pdf.apply(
+        lambda x: [x["data"][i] for i in x["idx"]]
+        if x["data"] is not None
+        else None,
+        axis=1,
+    )
```

Comment:
Feels a bit unnecessary to wrap this in a DataFrame. Couldn't we use the `idx` list of lists directly in a `Series.apply`?

Reply:
`Series.apply` does not provide the row-index inside ufunc, I think?
```python
def ufunc(row, idx):
    return [row[i] for i in idx[__row_num__]] #??

ps = pd.Series([[0, 1], [2, 3]])
idx = pd.Series([[1], [0]])
ps.apply(ufunc, args=(idx,))
```

A possible simpler way is we can zip the row and the index:
```python
ps = pd.Series([[0, 1], [2, 3]])
idx = pd.Series([[1], [0]])

pps = pd.Series(zip(ps, idx))
pps.map(lambda x: [x[0][i] for i in x[1]])
```

Reply:
Sorry I got this completely wrong. What you have is good.